stan/math/prim/mat/fun/gp_exponential_cov.hpp

```diff
@@ -0,0 +1,253 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_GP_EXPONENTIAL_COV_HPP
+#define STAN_MATH_PRIM_MAT_FUN_GP_EXPONENTIAL_COV_HPP
+
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/scal/fun/square.hpp>
+#include <stan/math/prim/scal/fun/squared_distance.hpp>
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <cmath>
+#include <vector>
+
+namespace stan {
+namespace math {
+
+/** Returns a Matern exponential covariance matrix with one input vector
+ *
+ * \f[ k(x, x') = \sigma^2 exp(-\frac{d(x, x')}{l}) \f]
+ *
+ * where \f$ d(x, x') \f$ is the squared distance, or the dot product.
+ *
+ * See Rausmussen & Williams et al 2006 Chapter 4.
+ *
+ * @param x std::vector of elements that can be used in stan::math::distance
+ * @param length_scale length scale
+ * @param sigma standard deviation that can be used in stan::math::square
+ * @throw std::domain error if sigma <= 0, l <= 0, or x is nan or inf
+ *
+ */
+template <typename T_x, typename T_s, typename T_l>
+inline typename Eigen::Matrix<typename stan::return_type<T_x, T_s, T_l>::type,
+                              Eigen::Dynamic, Eigen::Dynamic>
+gp_exponential_cov(const std::vector<T_x> &x, const T_s &sigma,
+                   const T_l &length_scale) {
+  using stan::math::square;
+  using stan::math::squared_distance;
+  using std::exp;
+  using std::pow;
+
+  size_t x_size = x.size();
+  for (size_t n = 0; n < x_size; ++n)
+    check_not_nan("gp_exponential_cov", "x", x[n]);
+
+  check_positive("gp_exponential_cov", "marginal variance", sigma);
+  check_not_nan("gp_exponential_cov", "marginal variance", sigma);
```

Comment:
use `check_positive_finite()` instead of `check_positive()` and `check_not_nan()`